syntax = "proto3";
package com.cloudscreen.log; 
option java_package = "com.cloudscreen.csg.log.pb";
option java_outer_classname = "CloudPolicyLogPB";
message CloudPolicyLog 
{ 
  uint64  matched_policy_id = 10;
  string  policy_name = 20;  
  
  uint32 log_type = 30;
  uint32 delivery_type = 40;
  uint32 severity = 50;
  
  uint64 enterprise_id = 52;
  uint64 user_id = 55;
  
  uint64 device_activity_id = 60;
  uint64 csp_product_id = 70;
  string csp_product_name = 80; 
  
  uint64 event_id = 90;
  uint64 event_category_id = 91;
  string event_name = 100; 
  
  string  session_id = 110;
  string  traffic_id = 120;
  
  uint64 matched_device_id = 130;
  string matched_device_name = 140;
  uint64 matched_map_id = 150;
  string matched_map_name = 160;
  uint64 matched_net_id = 170;
  string matched_net_name = 180;
 
  uint64 frequency_id = 182;
  string frequency_name = 183;
  uint32 threshold = 185;
  uint32 period = 187;
  uint32 period_unit = 189;
  
  // reserved field
  string policy = 190;
  
  //format - "yyyy-MM-dd HH:mm:ss", 24h timezone as UTC 
  string  trigger_time = 200;  	
  //
  uint32 action_taken = 210;
  string  comment = 220;  
  
  uint64 reserve_id1 = 230;
  string reserve_content1 = 240; 	
  uint64 reserve_id2 = 250;
  string reserve_content2 = 260;  
}

